cmd/oras/manifest/fetch.go

```diff
@@ -72,46 +82,93 @@ Example - Fetch manifest with prettified json result:
 	}
 
-	cmd.Flags().BoolVarP(&opts.pretty, "pretty", "", false, "output prettified manifest")
-	cmd.Flags().BoolVarP(&opts.fetchDescriptor, "descriptor", "", false, "fetch a descriptor of the manifest")
 	cmd.Flags().StringSliceVarP(&opts.mediaTypes, "media-type", "", nil, "accepted media types")
+	cmd.Flags().StringVarP(&opts.outputPath, "output", "o", "", "output file path")
 	option.ApplyFlags(&opts, cmd.Flags())
 	return cmd
 }
 
-func fetchManifest(opts fetchOptions) error {
+func fetchManifest(opts fetchOptions) (fetchErr error) {
 	ctx, _ := opts.SetLoggerLevel()
-	targetPlatform, err := opts.Parse()
-	if err != nil {
-		return err
-	}
+
 	repo, err := opts.NewRepository(opts.targetRef, opts.Common)
 	if err != nil {
 		return err
 	}
+
 	if repo.Reference.Reference == "" {
 		return oerrors.NewErrInvalidReference(repo.Reference)
 	}
 	repo.ManifestMediaTypes = opts.mediaTypes
 
-	// Fetch and output
-	var content []byte
-	if opts.fetchDescriptor {
-		content, err = cas.FetchDescriptor(ctx, repo, opts.targetRef, targetPlatform)
-	} else {
-		content, err = cas.FetchManifest(ctx, repo, opts.targetRef, targetPlatform)
-	}
+	targetPlatform, err := opts.Parse()
 	if err != nil {
 		return err
 	}
-	if opts.pretty {
-		buf := bytes.NewBuffer(nil)
-		if err = json.Indent(buf, content, "", "  "); err != nil {
-			return fmt.Errorf("failed to prettify: %w", err)
+
+	var src oras.ReadOnlyTarget = repo.Manifests()
+	if opts.cacheRoot != "" {
+		ociStore, err := oci.New(opts.cacheRoot)
+		if err != nil {
+			return err
 		}
-		buf.WriteByte('\n')
-		content = buf.Bytes()
+		src = cache.New(src, ociStore)
 	}
-	_, err = os.Stdout.Write(content)
-	return err
+
+	var desc ocispec.Descriptor
+	var content []byte
+	if opts.OutputDescriptor && opts.outputPath == "" {
+		// fetch manifest descriptor only
+		desc, err = oras.Resolve(ctx, src, opts.targetRef, oras.DefaultResolveOptions)
+		if err != nil {
+			return err
+		}
+	} else {
+		// fetch manifest content
+		desc, content, err = oras.FetchBytes(ctx, src, opts.targetRef, oras.FetchBytesOptions{
+			FetchOptions: oras.FetchOptions{
+				ResolveOptions: oras.ResolveOptions{
+					TargetPlatform: targetPlatform,
+				},
+			},
+			MaxBytes: 0,
+		})
+		if err != nil {
+			return err
+		}
+
+		// outputs manifest content
+		if opts.outputPath == "" || opts.outputPath == "-" {
+			return opts.Output(os.Stdout, content)
+		}
+
+		// save manifest content into the local file if the output path is provided
+		file, err := os.Create(opts.outputPath)
+		if err != nil {
+			return err
+		}
+		defer func() {
+			if err := file.Close(); fetchErr == nil {
+				fetchErr = err
+			}
+		}()
+
+		if _, err = file.Write(content); err != nil {
+			return err
+		}
+	}
+
+	// output manifest's descriptor if `--descriptor` is used
+	if opts.OutputDescriptor {
+		descBytes, err := json.Marshal(desc)
+		if err != nil {
+			return err
+		}
+		err = opts.Output(os.Stdout, descBytes)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
This code block appears everywhere if `--descriptor` and `--pretty` and both needed, any idea to refactor it? It doesn't necessarily need to be done in this PR /cc @lizMSFT

Reply:
> 

We're discussing about the change offline.